### ISSUE TITLE
🚨 Error: qc(...).finally is not a function

### ISSUE THREAD

**Issue by ampprojectbot** [ampproject/amphtml#27956]
Details
---
**Error report:** [link](http://go/ampe/CJ7CzPa70KW3fA)
**First seen:** Apr 8, 2020
**Frequency:** ~ 2,383/day

Stacktrace
---
```
Error: qc(...).finally is not a function
    at (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/extensions/amp-next-page/1.0/service.js:230)
    at build (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/extensions/amp-next-page/1.0/amp-next-page.js:42)
    at buildCallback (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/custom-element.js:478)
    at (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/custom-element.js:475)
    at build (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/service/resource.js:336)
    at resource (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/service/resources-impl.js:539)
    at buildResourceUnsafe_ (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/service/resources-impl.js:474)
    at buildOrScheduleBuildForResource_ (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/service/resources-impl.js:592)
    at upgraded (https://raw.githubusercontent.com/ampproject/amphtml/2004172112280/src/custom-element.js:384)
```

Notes
---
`@wassgha` modified `extensions/amp-next-page/1.0/service.js:230-235` in #26841 (Feb 28, 2020)
`@wassgha` modified `extensions/amp-next-page/1.0/amp-next-page.js:42-45` in #25636 (Dec 10, 2019)
`@zhouyx` modified `src/custom-element.js:477-479` in #14306 (Apr 3, 2018)
`@rsimha` modified `src/custom-element.js:475-475` in #21212 (May 16, 2019)
`@rsimha` modified `src/service/resource.js:336-338` in #21212 (May 16, 2019)
`@lannka` modified `src/service/resources-impl.js:537-542` in #23345 (Jul 17, 2019)
`@choumx` modified `src/service/resources-impl.js:474-474` in #27657 (Apr 9, 2020)
`@mkhatib` modified `src/service/resources-impl.js:592-592` in #4476 (Aug 12, 2016)
`@jridgewell` modified `src/custom-element.js:384-384` in #8229 (Mar 28, 2017)

**Comment by rcebulko** [comment]
@wassgha for awareness/in case there's an obvious culprit you recognize here. Frequency is pretty low though, so if nothing stands out this can probably be backlogged

**Comment by wassgha** [comment]
From a quick investigation, it seems like `Promise.finally()` is not available in some old version of Firefox, I'll defer to @ampproject/wg-runtime to check our polyfills (or whether we should ban usage of `.finally` if necessary).

**Comment by wassgha** [comment]
@jridgewell feel free to reassign

**Comment by jridgewell** [comment]
I think we should ban use of `finally`, it's an ES2019 feature. The polyfill we have is just to polyfill ES2015 to ES5, not the more recent features.

**Comment by jridgewell** [comment]
In fact, it's already banned: https://github.com/ampproject/amphtml/blob/97099ce8bf4826a08d964e7a02da55a7641b1090/build-system/test-configs/conformance-config.textproto#L186-L190

**Comment by wassgha** [comment]
@jridgewell how is that enforced? the PRs seem to pass

**Comment by jridgewell** [comment]
I think I made a mistake, it should be `BANNED_PROPERTY_CALL`. Looking into it now.

**Comment by wassgha** [comment]
Sounds good and I'll send a PR to remove usages

**Comment by rcebulko** [comment]
Thanks for fixing and preventing this y'all!
